apps/api/e2e/utils/server.ts

```diff
@@ -198,35 +95,28 @@ async function waitForServerReady(
       if (response.status === 200) {
         console.log("✅ Server is ready");
 
-        // Additional verification of API endpoints and metrics server
-        try {
-          // Wait a bit more to ensure all routes are registered
-          await new Promise((resolve) => setTimeout(resolve, 1000));
-
-          // Verify admin setup endpoint is available
-          const adminSetupResponse = await axios.get(`${BASE_URL}/api/health`);
-          console.log(
-            `Admin API health check: ${adminSetupResponse.status === 200 ? "OK" : "Failed"}`,
-          );
-
-          // Verify metrics server is available (always on localhost for security)
-          const metricsPort = process.env.METRICS_PORT || "3003";
-          const metricsUrl = `http://127.0.0.1:${metricsPort}`;
-          const metricsHealthResponse = await axios.get(`${metricsUrl}/health`);
-          console.log(
-            `Metrics server health check: ${metricsHealthResponse.status === 200 ? "OK" : "Failed"}`,
-          );
-        } catch (err) {
-          console.warn(
-            "Additional API/metrics verification failed, but continuing:",
-            err,
-          );
-        }
+        // Verify admin setup endpoint is available
+        const adminSetupResponse = await axios.get(`${BASE_URL}/api/health`);
+        console.log(
+          `Admin API health check: ${adminSetupResponse.status === 200 ? "OK" : "Failed"}`,
+        );
+
+        // Verify metrics server is available (always on localhost for security)
+        const metricsPort = process.env.METRICS_PORT || "3003";
+        const metricsUrl = `http://127.0.0.1:${metricsPort}`;
+        const metricsHealthResponse = await axios.get(`${metricsUrl}/health`);
+        console.log(
+          `Metrics server health check: ${metricsHealthResponse.status === 200 ? "OK" : "Failed"}`,
+        );
 
         return;
       }
-      // eslint-disable-next-line @typescript-eslint/no-unused-vars
     } catch (error) {
+      // @ts-expect-error - error is unknown type from catch block
+      if (error?.code !== "ECONNREFUSED") {
```

Comment:
`ECONNREFUSED` will be the code if the server isn't ready yet.  Otherwise we can throw the error and stop the tests.